docs/elasticsearch/index-templates.md

```diff
@@ -0,0 +1,204 @@
+---
+layout: default
+title: Index Templates
+parent: Elasticsearch
+nav_order: 5
+---
+
+# Index template
+
+An index template formats all newly created indexes to a one like you may already have by automatically picking up on a naming schema.
+
+For example, if you have log indices being continuously indexed every day, you can define an index template to have all these log indices with the same mappings and settings.
+
+---
+
+#### Table of contents
+1. TOC
+{:toc}
+
+
+---
+
+## Create template
+
+To create an index template, use a POST request:
+
+```json
+POST _template
+```
+
+This command creates a template named `daily_logs` and applies it to any new index that matches the regular expression `logs-2020-01-*` and also adds it to the `my_logs` alias:
+
+```json
+PUT _template/daily_logs
+{
+  "aliases": {
+    "my_logs": {}
+  },
+  "index_patterns": [
+    "logs-2020-01-*"
+  ],
+  "settings": {
+    "number_of_shards": 2,
+    "number_of_replicas": 1
+  },
+  "mappings": {
+    "properties": {
+      "timestamp": {
+        "type": "date",
+        "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+      },
+      "value": {
+        "type": "double"
+      }
+    }
+  }
+}
+```
+
+You should see the following response:
+
+```json
+{
+  "acknowledged": true
+}
+```
+
+Now, if you create an index named `logs-2020-01-01`:
+
+```json
+PUT logs-2020-01-01
+```
+
+It will have the mappings and settings defined in the `daily_logs` template:
+
+```json
+GET logs-2020-01-01
+```
+
+```json
+{
+  "logs-2020-01-01": {
+    "aliases": {
+      "my_logs": {}
+    },
+    "mappings": {
+      "properties": {
+        "timestamp": {
+          "type": "date",
+          "format": "yyyy-MM-dd HH:mm:ss||yyyy-MM-dd||epoch_millis"
+        },
+        "value": {
+          "type": "double"
+        }
+      }
+    },
+    "settings": {
+      "index": {
+        "creation_date": "1578107970779",
+        "number_of_shards": "2",
+        "number_of_replicas": "1",
+        "uuid": "U1vMDMOHSAuS2IzPcPHpOA",
+        "version": {
+          "created": "7010199"
+        },
+        "provided_name": "logs-2020-01-01"
+      }
+    }
+  }
+}
+```
+
+You can keep creating indexes: `logs-2020-01-02`, `logs-2020-01-03`, and so on, and they will all have the same mappings and settings.
```

Comment:
Any additional indices that match this pattern---`logs-2020-01-02`, `logs-2020-01-03`, etc.---will inherit the same mappings and settings.